pkg/modelprovider/provider.go

```diff
@@ -0,0 +1,46 @@
+/*
+ *     Copyright 2025 The CNAI Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package modelprovider
+
+import "context"
+
+// Provider defines the interface that all model providers must implement.
+// A provider is responsible for downloading models from a specific source
+// (e.g., HuggingFace, ModelScope, Civitai, etc.)
+type Provider interface {
+	// Name returns the human-readable name of the provider
+	// Example: "huggingface", "modelscope", "civitai"
+	Name() string
+
+	// SupportsURL checks if this provider can handle the given model URL
+	// This enables automatic provider detection based on URL patterns
+	SupportsURL(url string) bool
```

Comment:
I prefer to use explicit statement for this part, such as by scheme provided by user, which is more simple and clear, e.g `--model-url hf://meta-llama/Llama-3.3-70B-Instruct`